### DISCUSSION TITLE
Wix Toolset 3.11 using HeatDirectory task kills Visual Studio 2022

### DISCUSSION THREAD

**Discussion by nikryden** [wixtoolset/issues#7775] [Questions] [answered]
For me VS2022 was crashing when using Wix Toolset 3.11 after updated VS2022 to 17.1.6

**The error I was getting:**  
```
Could not load file or assembly 'file:///C:\Program Files (x86)\WiX Toolset v3.11\bin\Heat.exe' or one of its dependencies. An attempt was made to load a program with an incorrect format...
```

**To fix it try:**
In the msi wix project file under target add ```RunAsSeparateProcess="$(RunWixToolsOutOfProc)"``` to the HeatDirectory
```<HeatDirectory RunAsSeparateProcess="$(RunWixToolsOutOfProc)" OutputFile=...```

I found the solution in this issue from 2021 (https://github.com/wixtoolset/issues/issues/6636)

**Comment by robmen** [comment, marked as answer]
Yes, the solution is #6636
